src/test/java/uk/gov/companieshouse/api/accounts/validation/CreditorsAfterOneYearValidatorTest.java

```diff
@@ -0,0 +1,509 @@
+                UNEXPECTED_DATA_VALUE);
+
+        errors = validator.validateCreditorsAfterOneYear(creditorsAfterOneYear, mockTransaction, COMPANY_ACCOUNTS_ID, mockRequest);
+
+        assertTrue(errors.hasErrors());
+        assertEquals(2, errors.getErrorCount());
+        assertTrue(errors.containsError(createError(UNEXPECTED_DATA_VALUE,
+                CREDITORS_AFTER_CURRENT_PERIOD_PATH)));
+        assertTrue(errors.containsError(createError(UNEXPECTED_DATA_VALUE,
+                CREDITORS_AFTER_PREVIOUS_PERIOD_PATH)));
+    }
+
+    @Test
+    @DisplayName("Errors returned when no totals provided")
+    void testErrorsReturnedWhenNoTotalsProvided() throws ServiceException,
+            DataException {
+
+        creditorsAfterOneYear.setCurrentPeriod(new CurrentPeriod());
+        creditorsAfterOneYear.setPreviousPeriod(new PreviousPeriod());
+
+        mockValidBalanceSheetCurrentPeriod();
+        mockValidBalanceSheetPreviousPeriod();
+
+        ReflectionTestUtils.setField(validator, MANDATORY_ELEMENT_MISSING_NAME, MANDATORY_ELEMENT_MISSING_VALUE);
+        ReflectionTestUtils.setField(validator, CURRENT_BALANCE_SHEET_NOT_EQUAL_NAME,
+                CURRENT_BALANCE_SHEET_NOT_EQUAL_VALUE);
+        ReflectionTestUtils.setField(validator, PREVIOUS_BALANCE_SHEET_NOT_EQUAL_NAME,
+                PREVIOUS_BALANCE_SHEET_NOT_EQUAL_VALUE);
+
+        when(mockCompanyService.isMultipleYearFiler(mockTransaction)).thenReturn(true);
+
+        errors = validator.validateCreditorsAfterOneYear(creditorsAfterOneYear, mockTransaction, COMPANY_ACCOUNTS_ID, mockRequest);
+
+        assertTrue(errors.hasErrors());
+        assertEquals(4, errors.getErrorCount());
+        assertTrue(errors.containsError(createError(MANDATORY_ELEMENT_MISSING_VALUE,
+                CREDITORS_AFTER_CURRENT_PERIOD_TOTAL_PATH)));
+        assertTrue(errors.containsError(createError(CURRENT_BALANCE_SHEET_NOT_EQUAL_VALUE,
+                CREDITORS_AFTER_CURRENT_PERIOD_TOTAL_PATH)));
+        assertTrue(errors.containsError(createError(MANDATORY_ELEMENT_MISSING_VALUE,
+                CREDITORS_AFTER_PREVIOUS_PERIOD_TOTAL_PATH)));
+        assertTrue(errors.containsError(createError(PREVIOUS_BALANCE_SHEET_NOT_EQUAL_VALUE,
+                CREDITORS_AFTER_PREVIOUS_PERIOD_TOTAL_PATH)));
+    }
+
+    @Test
+    @DisplayName("Data exception thrown when company service API call fails")
+    void testDataExceptionThrown() throws ServiceException {
+
+        when(mockCompanyService.isMultipleYearFiler(mockTransaction)).thenThrow(mockServiceException);
+
+        assertThrows(DataException.class,
+                () -> validator.validateCreditorsAfterOneYear(creditorsAfterOneYear,
+                        mockTransaction, COMPANY_ACCOUNTS_ID, mockRequest));
+    }
+
+    @Test
+    @DisplayName("Data exception thrown when mongo current balancesheet call fails")
+    void testDataExceptionThrownWhenCurrentBalanceSheetMongoCallFails() throws ServiceException,
```

Comment:
Ideally we shouldn't include 'Mongo' in test names; this test isn't concerned with the database